pkg/armhelpers/azureclient.go

```diff
@@ -232,6 +232,34 @@ func NewAzureClientWithClientCertificate(env azure.Environment, subscriptionID,
 	return getClient(env, subscriptionID, tenantID, armSpt, graphSpt), nil
 }
 
+// NewAzureClientWithClientCertificateExternalTenant returns an AzureClient via client_id and jwt certificate assertion against a 3rd party tenant
+func NewAzureClientWithClientCertificateExternalTenant(env azure.Environment, subscriptionID, tenantID, clientID string, certificate *x509.Certificate, privateKey *rsa.PrivateKey) (*AzureClient, error) {
+	oauthConfig, err := adal.NewOAuthConfig(env.ActiveDirectoryEndpoint, tenantID)
```

Comment:
do we need to call this from getAuthArgs as well? Perhaps add a new auth method named "external-tenant".

Reply:
I'm confused. What are you suggesting?

Reply:
Referring to this : https://github.com/Azure/acs-engine/blob/e93da3756bc9b8a647b5771b08524915bcf34d70/cmd/root.go#L193